src/pudl/analysis/record_linkage/embed_dataframe.py

```diff
@@ -69,37 +73,36 @@ def as_pipeline(self):
         )
 
 
-def dataframe_embedder_factory(vectorizers: dict[str, ColumnVectorizer]):
-    """Return a configured op graph to embed an input dataframe."""
-
-    @op
-    def train_dataframe_embedder(df: pd.DataFrame):
-        """Train :class:`sklearn.compose.ColumnTransformer` on input."""
-        column_transformer = ColumnTransformer(
-            transformers=[
-                (name, column_transform.as_pipeline(), column_transform.columns)
-                for name, column_transform in vectorizers.items()
-            ],
-            transformer_weights={
-                name: column_transform.weight
-                for name, column_transform in vectorizers.items()
-            },
-        )
+@op
+def train_dataframe_embedder(
+    df: pd.DataFrame, vectorizers: dict[str, ColumnVectorizer]
+):
+    """Train :class:`sklearn.compose.ColumnTransformer` on input."""
+    column_transformer = ColumnTransformer(
+        transformers=[
+            (name, column_transform.as_pipeline(), column_transform.columns)
+            for name, column_transform in vectorizers.items()
+        ],
+        transformer_weights={
+            name: column_transform.weight
+            for name, column_transform in vectorizers.items()
+        },
+    )
 
-        return column_transformer.fit(df)
+    return column_transformer.fit(df)
 
-    @op
-    def apply_dataframe_embedder(df: pd.DataFrame, transformer: ColumnTransformer):
-        """Use :class:`sklearn.compose.ColumnTransformer` to transform input."""
-        return FeatureMatrix(matrix=transformer.transform(df))
 
-    @graph
-    def embed_dataframe(df: pd.DataFrame) -> FeatureMatrix:
-        """Train dataframe embedder and apply to input df."""
-        transformer = train_dataframe_embedder(df)
-        return apply_dataframe_embedder(df, transformer)
+@op
+def apply_dataframe_embedder(df: pd.DataFrame, transformer: ColumnTransformer):
+    """Use :class:`sklearn.compose.ColumnTransformer` to transform input."""
+    return FeatureMatrix(matrix=transformer.transform(df), index=df.index)
 
-    return embed_dataframe
+
+@graph
+def embed_dataframe_graph(df: pd.DataFrame, vectorizers) -> FeatureMatrix:
```

Comment:
Maybe `embed_dataframe_graph` isn't the best name for this graph. It was previously just `embed_dataframe` but now that you actually call this graph instead of just the `dataframe_embedder_factory`, I wanted to make sure that this graph did't ahve the same name as the module (`embed_dataframe`).